File: src/me/seeking/module/render/KeyDisplay.java
package me.seeking.module.render;

import me.seeking.event.EventTarget;
import me.seeking.event.events.EventRender2D;
import me.seeking.module.Module;
import me.seeking.ui.font.FontLoaders;
import me.seeking.utils.BlurUtil;
import me.seeking.utils.RenderUtil;
import net.minecraft.client.gui.Gui;
import org.lwjgl.input.Keyboard;

import java.awt.*;

/**
 * This file is a part of Seeking Client.
 * @author MiLiBlue
 */
public class KeyDisplay extends Module {
    public KeyDisplay(){
        super("KeyDisplay", ModuleType.Render);
        setX(0);
        setY(40);
        setWidth(63);
        setHeight(41);
    }

    @EventTarget
    public void render2D(EventRender2D e){
        //Forward
        RenderUtil.drawShadow((float) (getX() + 21), (float) getY(), 20, 20);
        RenderUtil.drawRect((float) (getX()+21), (float) getY(), (float) (getX()+20+21), (float) (getY()+20),  mc.gameSettings.keyBindForward.isKeyDown()? new Color(255, 255, 255, 135).getRGB() : new Color(0, 0, 0, 135).getRGB());
        FontLoaders.font16.drawString(Keyboard.getKeyName(mc.gameSettings.keyBindForward.getKeyCode()).toUpperCase(), (float) (getX() + 26.5), (float) (getY()+6), -1);
        //BlurUtil.blurArea((float) (getX()+21), (float) getY(), 20, 20);
        //Right
        RenderUtil.drawShadow((float) (getX() + 42), (float) (getY()+21), 20, 20);
        RenderUtil.drawRect((float) (getX()+42), (float) (getY()+21), (float) (getX()+20+42), (float) (getY()+20+21),  mc.gameSettings.keyBindRight.isKeyDown()? new Color(255, 255, 255, 135).getRGB() : new Color(0, 0, 0, 135).getRGB());
        FontLoaders.font16.drawString(Keyboard.getKeyName(mc.gameSettings.keyBindRight.getKeyCode()).toUpperCase(), (float) (getX() + 49), (float) (getY()+27), -1);
        //BlurUtil.blurArea((float) (getX() + 42), (float) (getY()+21), 20, 20);
        //Left
        RenderUtil.drawShadow((float) getX(), (float) getY()+21, 20, 20);
        RenderUtil.drawRect((float) getX(), (float) (getY()+21), (float) (getX()+20), (float) (getY()+20+21), mc.gameSettings.keyBindLeft.isKeyDown()? new Color(255, 255, 255, 135).getRGB() : new Color(0, 0, 0, 135).getRGB());
        FontLoaders.font16.drawString(Keyboard.getKeyName(mc.gameSettings.keyBindLeft.getKeyCode()).toUpperCase(), (float) (getX() + 7), (float) (getY()+27), -1);
        //BlurUtil.blurArea((float) getX(), (float) getY()+21, 20, 20);
        //Back
        RenderUtil.drawShadow((float) (getX() + 21), (float) getY()+21, 20, 20);
        RenderUtil.drawRect((float) (getX()+21), (float) (getY()+21), (float) (getX()+20+21), (float) (getY()+20+21), mc.gameSettings.keyBindBack.isKeyDown()? new Color(255, 255, 255, 135).getRGB() : new Color(0, 0, 0, 135).getRGB());
        FontLoaders.font16.drawString(Keyboard.getKeyName(mc.gameSettings.keyBindBack.getKeyCode()).toUpperCase(), (float) (getX() + 28.5), (float) (getY()+27), -1);
        //BlurUtil.blurArea((float) (getX() + 21), (float) getY()+21, 20, 20);
    }
}

File: src/main/java/me/seeking/module/render/KeyDisplay.kt
package me.seeking.module.render

import me.seeking.event.EventTarget
import me.seeking.event.events.EventRender2D
import me.seeking.module.Module
import me.seeking.ui.font.FontLoaders
import me.seeking.utils.RenderUtil
import net.minecraft.client.gui.Gui
import org.lwjgl.input.Keyboard
import java.awt.Color

/**
 * This file is a part of Seeking Client.
 * @author MiLiBlue
 */
class KeyDisplay : Module("KeyDisplay", ModuleType.Render) {
    init {
        setX(0.0)
        setY(40.0)
        setWidth(63.0)
        setHeight(41.0)
    }

    @EventTarget
    fun render2D(e: EventRender2D?) {
        //Forward
        RenderUtil.drawShadow((getX() + 21).toFloat(), getY().toFloat(), 20F, 20F)
        RenderUtil.drawRect(
            (getX() + 21),
            getY(),
            (getX() + 20 + 21),
            (getY() + 20),
            if (mc.gameSettings.keyBindForward.isKeyDown) Color(255, 255, 255, 135).rgb else Color(0, 0, 0, 135).rgb
        )
        FontLoaders.font16!!.drawStringWithShadow(
            Keyboard.getKeyName(mc.gameSettings.keyBindForward.keyCode).toUpperCase(), (getX() + 26).toFloat()
                .toDouble(), (getY() + 7).toFloat().toDouble(), -1
        )
        //Right
        RenderUtil.drawShadow((getX() + 42).toFloat(), (getY() + 21).toFloat(), 20F, 20F)
        RenderUtil.drawRect(
            (getX() + 42),
            (getY() + 21),
            (getX() + 20 + 42),
            (getY() + 20 + 21),
            if (mc.gameSettings.keyBindRight.isKeyDown) Color(255, 255, 255, 135).rgb else Color(0, 0, 0, 135).rgb
        )
        FontLoaders.font16!!.drawStringWithShadow(
            Keyboard.getKeyName(mc.gameSettings.keyBindRight.keyCode).toUpperCase(), (getX() + 48).toFloat()
                .toDouble(), (getY() + 28).toFloat().toDouble(), -1
        )
        //Left
        RenderUtil.drawShadow(getX().toFloat(), getY().toFloat() + 21, 20F, 20F)
        RenderUtil.drawRect(
            getX(),
            (getY() + 21),
            (getX() + 20),
            (getY() + 20 + 21),
            if (mc.gameSettings.keyBindLeft.isKeyDown) Color(255, 255, 255, 135).rgb else Color(0, 0, 0, 135).rgb
        )
        FontLoaders.font16!!.drawStringWithShadow(
            Keyboard.getKeyName(mc.gameSettings.keyBindLeft.keyCode).toUpperCase(), (getX() + 7).toFloat()
                .toDouble(), (getY() + 28).toFloat().toDouble(), -1
        )
        //Back
        RenderUtil.drawShadow((getX() + 21).toFloat(), getY().toFloat() + 21, 20F, 20F)
        RenderUtil.drawRect(
            (getX() + 21),
            (getY() + 21),
            (getX() + 20 + 21),
            (getY() + 20 + 21),
            if (mc.gameSettings.keyBindBack.isKeyDown) Color(255, 255, 255, 135).rgb else Color(0, 0, 0, 135).rgb
        )
        FontLoaders.font16!!.drawStringWithShadow(
            Keyboard.getKeyName(mc.gameSettings.keyBindBack.keyCode).toUpperCase(), (getX() + 28).toFloat()
                .toDouble(), (getY() + 28).toFloat().toDouble(), -1
        )
    }
}
